src/Schema/Factories/FieldFactory.php

```diff
@@ -224,10 +239,10 @@ public function getRules($rootValue, $inputArgs, $context, ResolveInfo $resolveI
             })
             ->filter();
 
-        // Rules are applied to the fields which are on the root operation types
-        // Nested fields are excluded because they are validated as part of the root field
+        // Rules are applied to the fields which are on the root Mutation type.
+        // Nested fields are excluded because they are validated as part of the root field.
         $parentOperationType = data_get($resolveInfo, 'parentType.name');
-        if ('Mutation' === $parentOperationType || 'Query' === $parentOperationType) {
+        if ('Mutation' === $parentOperationType) {
```

Comment:
The difference between Query and Mutation is only a semantic one, i see no need to treat them different here.

Reply:
The reason is because we don't want the entire query to fail if a nested field's rule fails. For instance

```graphql
{
  me {
    first_name
    posts(invalidArg: "foo") {
      title
    }
  }
}
```

This should return the `me` fields `first_name` field and `null` for the `posts` field w/ validation meta. If we ran rules on the entire query then we would get `null` back for `me`.

Reply:
That should not happen, definitely agree with that. However, i think excluding Queries from Validation is not the best way to achieve that.

Reply:
Not completely done w/ the PR, the Query's root fields will be validated... just not the nested fields like mutations.

Reply:
But it needs to be handled differently, so there will be an `else if` statement here

Reply:
Mutations and Queries are absolutely the same thing. The only difference is that Queries carry the semantics of not changing stuff on the server.

Reply:
I would not even begin to handle them any different, because it might lead us down a path where they diverge more and more.

Reply:
Correct me if i am wrong, but to me it looks as if before this part of the code did handle fields the same whether they are defined on the parent type "Mutation" or "Query". Now, you removed the condition where it includes "Query".

Reply:
Right, because if we run rules the same way for both Queries and Mutations then an entire query will fail if a field that is nested 10 levels deep didn't get the correct args which is not what we want. So in other words, we want to handle Queries and Mutations differently.

Reply:
Another way to think about is Mutations have input types while queries have object types, and they need to be handled differently because an input type is not the same thing as an object type.

Reply:
That could easily happen with mutations too. There might be some args down below in the fields that are requested back from the server that might be invalid. You are right in that this should not cause the whole operation to fail.

Reply:
Great use case, if the user has a valid mutation but an invalid arg in the fields requested back that shouldn't cause the mutation to fail.

I'm currently writing tests for this PR and I'll make sure that is one of the checks.

Reply:
While Queries and Mutations are used for different purposes, they are structurally the same. They tend to look a bit different, mutations are usually more heavy on InputArguments while queries tend to have more nested fields - but in the end, both are just Object Types.

Reply:
I think we can not lump the fields of all args together in one large validation block, but should rather run one validation per field. Each nested sub-field gets its own validation context, and if that fails, only that part of the operation should be nulled.

Reply:
Can input type fields have args?

Reply:
Nope. From a type system perspective, they are actually the same thing as field args.

Reply:
>I think we can not lump the fields of all args together in one large validation block, but should rather run one validation per field. Each nested sub-field gets its own validation context, and if that fails, only that part of the operation should be nulled.

I don't feel the same way here. Mutation should fail (as they currently do) if the *input* provided is invalid (as Laravel would). On a query, only fields that have invalid rules/args should be returned as null (since they have a resolver) and validation message(s) attached to the output.

Reply:
````graphql

Reply:
However, if the root mutation had an input that is a large nested InputObject, and any of that is invalid, then i would fail the whole mutation.

Reply:
Correct... the mutation args are valid... the *ObjectType's* `bar` field has an incorrect arg and would return `null`.

Reply:
Yup. And we could just go ahead and replace `mutation` by `query` here, and execution-wise, nothing should change!

Reply:
Just so we're on the same page, that's how it currently works in this PR.

Reply:
```
-	    if ('Mutation' === $parentOperationType || 'Query' === $parentOperationType)
+       if ('Mutation' === $parentOperationType)